domain/telemetry.go

```diff
@@ -0,0 +1,54 @@
+package domain
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	// sqs_ingest_usecase_process_block_duration_ms
+	//
+	// histogram that measures the duration of processing a block in milliseconds in ingest usecase
+	//
+	// Has the following labels:
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockDurationMetricName = "sqs_ingest_usecase_process_block_duration_ms"
+
+	// sqs_ingest_usecase_process_block_error
+	//
+	// counter that measures the number of errors that occur during processing a block in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockErrorMetricName = "sqs_ingest_usecase_process_block_error"
+
+	// sqs_ingest_usecase_parse_pool_error
+	//
+	// counter that measures the number of errors that occur during pool parsing in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	SQSIngestUsecaseParsePoolErrorMetricName = "sqs_ingest_usecase_parse_pool_error"
```

Comment:
Note: linking to Linear task: https://linear.app/osmosis/issue/STABI-70/instrument-new-metrics-for-stage-2-sqs-grpc-refactor